tests/utilities_t/executors_t/test_sshexecutor.py

```diff
@@ -80,18 +103,16 @@ def test_establish_connection(self):
         self.mock_asyncssh.connect.side_effect = None
 
     def test_connection_property(self):
```

Comment:
Can we add an explicit test for probing the connections as well as the `bounded_connection` (queuing), please? If this is not much effort.

Reply:
I've added a test for probing (simple) and a test for queueing (vodoo). It would be good if you could check whether the latter is understandable/maintainable. (Bonus points if you can think of a simpler way.)

Reply:
I do not know if the following approach is better, because it is less prone to timing issues. Let's assume `MaxSessions` is 10. What about starting let's say ten tasks sleeping a long time ~(`async.sleep(5)`). Afterwards start a task sleeping ~0 ~`async.sleep(0.1)` and check that it does not return intime. Afterwards cancel all tasks using `task.cancel()`. Is this something that could be realized?

Reply:
That seems much simpler indeed. I'll give it a try.

Reply:
Yes, much simpler indeed...

Reply:
Bonus points, please! ;-)

Reply:
Here's a "👍" just for you: 👍